lib/commands/setup.js

```diff
@@ -21,180 +21,144 @@ class SetupCommand extends Command {
         yargs = StartCommand.configureOptions('start', yargs, extensions, true);
     }
 
-    constructor(ui, system) {
-        super(ui, system);
-
-        this.stages = [];
-    }
-
-    addStage(name, fn, dependencies, description) {
-        this.stages.push({
-            name: name,
-            description: description || name,
-            dependencies: dependencies && (Array.isArray(dependencies) ? dependencies : [dependencies]),
-            fn: fn
-        });
-    }
-
     run(argv) {
-        const os = require('os');
         const url = require('url');
         const path = require('path');
         const semver = require('semver');
+        const flatten = require('lodash/flatten');
 
         const linux = require('../tasks/linux');
         const migrator = require('../tasks/migrator');
         const configure = require('../tasks/configure');
 
-        if (argv.local) {
-            argv.url = argv.url || 'http://localhost:2368/';
-            argv.pname = argv.pname || 'ghost-local';
-            argv.process = 'local';
-            argv.stack = false;
-
-            // If the user's already specified a db client, then we won't override it.
-            if (!argv.db) {
-                argv.db = argv.db || 'sqlite3';
-                argv.dbpath = path.join(process.cwd(), 'content/data/ghost-local.db');
-            }
+        const {local = false, stages = []} = argv;
 
-            argv.start = (typeof argv.start === 'undefined') ? true : argv.start;
+        if (local) {
+            argv = Object.assign({
+                url: 'http://localhost:2368',
+                pname: 'ghost-local',
+                process: 'local',
+                stack: false,
+                db: 'sqlite3',
+                dbpath: argv.db ? null : path.join(process.cwd(), 'content/data/ghost-local.db'),
+                start: true
+            }, argv);
 
             // In the case that the user runs `ghost setup --local`, we want to make
             // sure we're set up in development mode
             this.system.setEnvironment(true, true);
         }
 
-        if (argv.stages && argv.stages.length) {
-            const instance = this.system.getInstance();
+        const instance = this.system.getInstance();
+
+        // This is used so we can denote built-in setup steps
+        // and disable the "do you wish to set up x?" prompt
+        // We use symbols so extension-defined tasks can't
+        // mark themselves as internal and skip the prompt
+        const internal = Symbol('internal setup step');
```

Comment:
as far as I can tell this is what symbols are _supposed_ to be used for, and it seemed like a good solution.